blockchain/blockchain.go

```diff
@@ -2021,6 +2021,13 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 		if bc.engine.CreateSnapshot(bc, block.NumberU64(), block.Hash(), nil) != nil {
 			return i, events, coalescedLogs, err
 		}
+
+		// update governance parameters
+		if istanbul, ok := bc.engine.(consensus.Istanbul); ok {
```

Comment:
@ian0371 Could you review that calling UpdateParam only when we use istanbul engine is okay?

Reply:
lgtm